galaxy/importer/linters/__init__.py

```diff
@@ -72,3 +72,29 @@ def _check_files(self, paths):
         for line in proc.stdout:
             yield line.strip()
         proc.wait()
+
+
+class AnsibleLinter(BaseLinter):
+
+    cmd = 'ansible-lint'
+
+    def _check_files(self, paths):
+        rules_folder = '/galaxy/galaxy/importer/galaxy-lint-rules/rules'
+
+        # different logic needed for multi role repos since
+        # ansible-lint issue role path cannot contain '/'
+        if paths == ['.']:  # single role repo
+            cmd = [self.cmd, '-p', '-r', rules_folder] + paths
+            logger.debug('CMD: ' + ' '.join(cmd))
+            proc = subprocess.Popen(cmd, cwd=self.root, stdout=subprocess.PIPE)
+        else:  # multi role repo
+            cmd = [self.cmd, '-p', '-r', rules_folder] + ['.']
+            logger.debug('CMD: ' + ' '.join(cmd))
+            cwd = '%s/%s' % (self.root, paths[0])
+            proc = subprocess.Popen(cmd, cwd=cwd, stdout=subprocess.PIPE)
```

Comment:
You can move `subprocess.Popen` and `logger.debug` outside if to reduce code duplication.

Reply:
Even more: only `cwd=` differs in these blocks.
(`if path == ['.']`, use `['.']`, else use same `['.']`, just not in var, for the last path part)

Reply:
Updated